source/userguide/auto_annotator.rst

```diff
@@ -0,0 +1,489 @@
+		"unannotate_supported_entities_only": True, 
+		"unannotate": [{"domains": ".*", "intents": ".*", "files": ".*", "entities": ".*"}],
+	}
+
+To unannotate you can run python code or use the command-line.
+
+Python:
+
+.. code-block:: python
+
+	from mindmeld.auto_annotator import SpacyAnnotator 
+	sa = SpacyAnnotator(app_path="hr_assistant")
+
+	sa.unannotate()
+
+Command-line:
+
+.. code-block:: console
+
+	mindmeld unannotate --app-path "hr_assistant"
+
+You can annotate in a similar manner.
+
+Python:
+
+.. code-block:: python
+
+	from mindmeld.auto_annotator import SpacyAnnotator 
+	sa = SpacyAnnotator(app_path="hr_assistant")
+
+	sa.annotate()
+
+Command-line:
+
+.. code-block:: console
+
+	mindmeld annotate --app-path "hr_assistant"
+
+The following section explains this same process in more detail.
+
+Using the Auto Annotator
+------------------------
+
+The Auto Annotator can be used by importing a class that implements the :class:`Annotator` abstract class in the :mod:`auto_annotator` module or through the command-line.
+We will demonstrate both approaches for unannotation and annotation using the :class:`SpacyAnnotator` class. We will first describe unannotation and then annotation. However, in a normal workflow you are likely to annotate first and then unannotate any annotations you are not pleased with.
+
+Unannotate
+^^^^^^^^^^
+To unannotate by creating an instance of the :class:`Annotator` class, run:
+
+.. code-block:: python
+
+	from mindmeld.auto_annotator import SpacyAnnotator 
+	sa = SpacyAnnotator(app_path="hr_assistant")
+
+	sa.unannotate()
+
+Alternatively, you can :attr:`unannotate` using the command-line:
+
+.. code-block:: console
+
+	mindmeld unannotate --app-path "hr_assistant"
+
+
+If you see the following message, you need to update the unannotate parameter in your custom :attr:`AUTO_ANNOTATOR_CONFIG` dictionary in :attr:`config.py`. You can refer to the config specifications in the section above.
+
+.. code-block:: console
+
+	'unannotate' is None in the config. Nothing to unannotate.
+
+
+Let's unannotate :attr:`sys_time` entities from the :attr:`get_date_range_aggregate` intent in the :attr:`date` domain.
+To do this, we can add the following :attr:`AUTO_ANNOTATOR_CONFIG` dictionary to :attr:`config.py`.
+
+
+.. code-block:: python
+
+	AUTO_ANNOTATOR_CONFIG = { 
+
+		"annotator_class": "SpacyAnnotator",
+		"overwrite": False, 
+		"annotate": [{"domains": ".*", "intents": ".*", "files": ".*", "entities": ".*"}],
+		"unannotate_supported_entities_only": True, 
+		"unannotate": [
+			{ 
+				"domains": "date", 
+				"intents": "get_date_range_aggregate", 
+				"files": "train.txt",
+				"entities": "sys_time", 
+			}
+		], 
+	}
+
+.. note::
+
+	The content of :attr:`annotate` in the config has no effect on unannotation. Similarly, :attr:`unannotate` in the config has no affect on annotation. These processes are independent and are only affected by the corresponding parameter in the config.
```

Comment:
What is the reason for having separate params for `annotate` and `unannotate`? should they be operating on the same files?

Reply:
It may not always be the case that you would want to unannotate and annotate the same entities. Often times you would like to annotate a set of entities and then unannotate a subset that you realised you would rather not have.

By having these as separate parameters, a developer doesn't have to copy and paste a different set of rules each time they switch methods and doesn't risk accidentally unannotating their existing annotations